### PR TITLE
admin activity log: wrong user time zone reported #4525

### LINKED ISSUE #4525: admin activity log: wrong user time zone reported
This user is from UTC-5 timezone
image

### PR DESCRIPTION
**Pull request by bamboo3250**
fix #4525

The issue happens only when the user is unregistered and his name is used in the query instead of his google id. Therefore, to solve this, I introduce "course" attribute to the query to indicate the course that unregistered user is belonged to. The query to search for activities of the user will be done via Search page by clicking on "View recent actions".

@kanghj could you review this?

### CODE REVIEW

**Comment by kanghj** [comment]
While it solves this particular issue, the "course" attribute does not work as advertised. If I search using "course" without an attribute for the user, then the results are the same as if the query was an empty query?

File: src/main/java/teammates/ui/controller/AdminActivityLogPageAction.java [hunk 1]
```diff
@@ -118,6 +118,12 @@ private void generateStatusMessage(AdminActivityLogPageData data, List<ActivityL
         if (data.isPersonSpecified()) {
             String targetUserGoogleId = data.getPersonSpecified();
             targetTimeZone = getLocalTimeZoneForRequest(targetUserGoogleId, "");
+            // if cannot find the user or the user is unregistered
```
Comment (kanghj) [inline src/main/java/teammates/ui/controller/AdminActivityLogPageAction.java]: I think this is an inaccurate comment, there are other reasons why the time zone will be uninitialised

**Comment by bamboo3250** [comment]
The course does not affect the filter but only the timezone shown to admin. Perhaps it's still better to put the course in `role`.

**Comment by bamboo3250** [comment]
@kanghj Now I removed `course` from the query and replaced by using POST action to pass `courseId` from Search Page to Activity Log Page. Please review again.

**Pull request merged**